sirch-inc/sirch-coin
# SIRCO-35

**Linear SIRCO-35**

This update adds a “Forgot Password” flow which can be found by navigating to Login > Forgot Password. It continues to use Supabase Auth, in this case to handle the sending of the password reset email and then passes a `“PASSWORD_RECOVERY”` event to the Update Password page. 

This includes the following new components:
* `ResetPassword.jsx` – allows a user to enter their email address and request a password reset. 
* `UpdatePassword.jsx` – allows a user to enter a new password for their account. 

To test the flow:
1. Navigate to the login page in the nav and select “Reset Password”
2. Enter the email that you signed up to Sirch Coins with and click “Reset Password”
3. You should get an email shortly from Supabase with a link to click to reset your password (alternatively, you may get a message that says we’ve hit the email rate limit, in which case you may need to wait to test again). 
4. Click the link to reset your password (you should be directed to the `/update-password` page).  
5. Enter a new password and click Update Password. 
6. If successful, you should get an `alert` confirming the password change and be redirected to the homepage. 
7. Logout from the nav
8. Try logging back in using your new password and confirm you can successfully log in.
